src/commands/src/helpers/dependency-helpers.ts

```diff
@@ -33,12 +34,45 @@ async function createEnvProdFile(repoDir: string, projectName: string, baseUrl:
   }
 }
 
-export async function prepareBuildAndGetDependencies(repoName: string, projectName: string, baseUrl: string) {
+export async function downloadBuildDependencies(buildDependencies: BuildDependencies, repoName: string) {
+  const commands: string[] = [];
+  buildDependencies.map(async (dependencyInfo) => {
```

Comment:
I believe with a map, this will return a Promise immediately, but the js thread won't actually await the completion of the promises. You could use Promise.all or allSettled to handle this asynchronicity!